app.js

```diff
@@ -1,66 +1,61 @@
-const fs = require("fs");
+const fs = require("fs/promises");
 const path = require("path");
-const StreamZip = require("node-stream-zip");
-const Archiver = require("./archiver");
-const { pathExists } = require("./helpers");
-
-async function zip(action) {
-  const pathToArchive = action.params.ARCHIVEPATH;
-  const pathToTargets = action.params.TARGETS;
-
-  const directoryPath = path.dirname(pathToArchive);
-  if (!await pathExists(directoryPath)) {
-    throw new Error(`Path ${directoryPath} does not exist!`);
-  }
-
-  const archiver = new Archiver(pathToArchive);
-
-  const targetsArray = pathToTargets.split("\n");
-  for (let i = 0; i < targetsArray.length; i += 1) {
-    archiver.add(targetsArray[i]);
+const { bootstrap } = require("kaholo-plugin-library");
+const {
+  pathExists,
+  createZipArchive,
+  unzipArchive,
+  validatePaths,
+} = require("./helpers");
+
+async function zip({
+  ARCHIVEPATH: archivePath,
+  TARGETS: targetPaths,
+  ignoredPaths,
+  overwrite,
+}) {
+  const absoluteArchivePath = path.resolve(archivePath);
+  const absoluteTargetPaths = targetPaths.map((targetPath) => path.resolve(targetPath));
+  const absoluteIgnoredPaths = (ignoredPaths || []).map((ignorePath) => path.resolve(ignorePath));
+  await validatePaths(...absoluteTargetPaths, path.dirname(absoluteArchivePath));
+
+  const archivePathExists = await pathExists(absoluteArchivePath);
+  if (overwrite && archivePathExists) {
+    await fs.rm(absoluteArchivePath, { recursive: true });
+  } else if (!overwrite && archivePathExists) {
+    throw new Error(`File ${archivePath} already exists. Please select the "Overwrite Existing Zip Archive" parameter to overwrite.`);
   }
```

Comment:
```suggestion
  if (!overwrite && archivePathExists) {
    throw new Error(`File ${archivePath} already exists. Please select the "Overwrite Existing Zip Archive" parameter to overwrite.`);
  }
  
  if (archivePathExists) {
    await fs.rm(absoluteArchivePath, { recursive: true });
  }
```

This way the same algorithmic complexity is maintained (1 condition evaluation at best, 2 at worst case), but we separate the functionalities - unhappy path from optional action.